net/web_relay_manager.go

```diff
@@ -0,0 +1,210 @@
+package net
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"encoding/json"
+	peer "gx/ipfs/QmYVXrKrKHDC9FobgmcmshCDyWwdrfwfanNQN4oxJ9Fk3h/go-libp2p-peer"
+	"time"
+
+	"github.com/OpenBazaar/openbazaar-go/pb"
+	"github.com/golang/protobuf/ptypes/any"
+
+	"gx/ipfs/QmerPMzPk1mJVowm8KgmoknWa4yCYvvugMPsgWmDNUvDLW/go-multihash"
+
+	"github.com/btcsuite/btcutil/base58"
+
+	"github.com/gorilla/websocket"
+)
+
+type WebRelayManager struct {
+	webrelays   []string
+	peerID      string
+	connections []*websocket.Conn
+	obService   NetworkService
+}
+
+type EncryptedMessage struct {
+	Message   string `json:"encryptedMessage"`
+	Recipient string `json:"recipient"`
+}
+
+type TypedMessage struct {
+	Type string
+	Data json.RawMessage
+}
+
+type SubscribeMessage struct {
+	UserID          string `json:"userID"`
+	SubscriptionKey string `json:"subscriptionKey"`
+}
+
+type SubscribeResponse struct {
+	Subscribe string `json:"subscribe"`
+}
+
+func NewWebRelayManager(webrelays []string, sender string) *WebRelayManager {
+	return &WebRelayManager{webrelays, sender, nil, nil}
+}
+
+func (wrm *WebRelayManager) ConnectToRelays(service NetworkService) {
+
+	// Set WRM service
+	wrm.obService = service
+
+	// Establish connections
+	var conns []*websocket.Conn
+	for _, relay := range wrm.webrelays {
+
+		// Connect and subscribe to websocket server
+		conn, err := wrm.connectToServer(relay, wrm.peerID)
+		if err != nil {
+			log.Error("Could not connect to: %s", relay)
+		}
```

Comment:
@hoffmabc `wrm.connections = append(conns, conn)` should be in an `else` bock of `if err != nil {` otherwise in the case of a connection error, your are adding a `nil` connection to the list.

Reply:
good catch